cookbooks/creating_a_behat_extension.rst

```diff
@@ -0,0 +1,270 @@
+While there might be alternatives, we will stick to the straightforward method.
+
+The ``getConfigKey`` method is used to identify our extension in the configuration, and the ``configure`` method is used to define the configuration tree.
+
+The code for `LogsExtension.php`:
+
+.. code-block:: php
+  
+  <?php
+  
+  namespace Behat\LogsExtension\ServiceContainer;
+  
+  use Behat\Testwork\ServiceContainer\Extension;
+  use Behat\Testwork\ServiceContainer\ExtensionManager;
+  use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+  use Symfony\Component\DependencyInjection\ContainerBuilder;
+  
+  class LogsExtension implements Extension
+  {
+      public function getConfigKey()
+      {
+          return 'logs_extension';
+      }
+  
+      public function initialize(ExtensionManager $extensionManager)
+      {
+          // Empty for our case, but useful to hook into other extensions' configurations
+      }
+
+      public function configure(ArrayNodeDefinition $builder)
+      {
+          $builder
+              ->addDefaultsIfNotSet()
+              ->children()
+                  ->scalarNode('enable')->defaultFalse()->end()
+                  ->scalarNode('filepath')->defaultValue('behat.log')->end()
+              ->end()
+          ;
+      }
+  
+      public function load(ContainerBuilder $container, array $config)
+      {
+          // ... we'll load our configuration here
+      }
+
+      public function process(ContainerBuilder $container)
+      {
+          // Empty for our case but needed for CompilerPassInterface
+      }
+  }
+
+.. note::
+  
+  The ``initialize`` and ``process`` methods are empty in our case but are useful when you need to interact with other extensions or process the container after it has been compiled.
+
+Initializing the Context
+------------------------
+
+To pass configuration values to our ``LogsContext``, we need to create an initializer.
+
+Directory structure:
+
+.. code-block::
+  src/
+      Context/
+          Initializer/
+              LogsInitializer.php   # This will handle context initialization
+          LogsContext.php
+      ServiceContainer/
+          LogsExtension.php
+
+The code for ``LogsInitializer.php``:
+
+.. code-block:: php  
+  <?php
+
+  namespace Behat\LogsExtension\Context\Initializer;
+
+  use Behat\LogsExtension\Context\LogsContext;
+  use Behat\Behat\Context\Context;
+  use Behat\Behat\Context\Initializer\ContextInitializer;
+
+  class LogsInitializer implements ContextInitializer
+  {
+      private string $filepath;
+      private bool $enable;
+  
+      public function __construct(string $filepath, bool $enable)
+      {
+          $this->filepath = $filepath;
+          $this->enable = $enable;
+      }
+  
+      public function initializeContext(Context $context)
+      {
+          if (!$context instanceof LogsContext) {
+              return;
+          }
```

Comment:
Might be useful to explain a little more what this method is? It was the "aha" moment for me when I started trying to write / use extensions...

```suggestion
          /* 
           * At the start of every scenario, behat will create a new instance of every `Context`
           * registered in your project. It will then call this method with each new `Context` in
           * turn. If you want to initialise multiple contexts, you can of course give them an 
           * interface and check for that here.
           */
          if (!$context instanceof LogsContext) {
              return;
          }
```